src/ralph/data_center/tests/test_admin.py

```diff
@@ -0,0 +1,184 @@
+from unittest import mock
+
+from django.contrib.auth import get_user_model
+from django.core import mail
+from django.db import connection, transaction
+from django.test import override_settings, RequestFactory, TransactionTestCase
+
+from ralph.accounts.tests.factories import UserFactory
+from ralph.assets.tests.factories import (
+    ServiceEnvironmentFactory,
+    ServiceFactory
+)
+from ralph.data_center.models import DataCenterAsset
+from ralph.data_center.tests.factories import DataCenterAssetFactory
+from ralph.lib.custom_fields.models import (
+    CustomField,
+    CustomFieldTypes,
+    CustomFieldValue
+)
+
+
+# TransactionTestCase has to be used here, since request to admin is wrapped
+# in transaction (to test publishing dc host update data to hermes)
+class DataCenterAssetAdminTest(TransactionTestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_superuser(
+            username='root',
+            password='password',
+            email='email@email.pl'
+        )
+        result = self.client.login(username='root', password='password')
+        self.assertEqual(result, True)
+        self.factory = RequestFactory()
+        self.dca = DataCenterAssetFactory(hostname='ralph1.allegro.pl')
+        self.custom_fields_inline_prefix = 'custom_fields-customfieldvalue-content_type-object_id-'  # noqa
+        self.custom_field_str = CustomField.objects.create(
+            name='test_str', type=CustomFieldTypes.STRING, default_value='xyz'
+        )
+        self.custom_field_choices = CustomField.objects.create(
+            name='test_choice', type=CustomFieldTypes.CHOICE,
+            choices='qwerty|asdfgh|zxcvbn', default_value='zxcvbn',
+            use_as_configuration_variable=True,
+        )
+
+    def _update_dca(self, dca_data=None, inline_data=None):
+        data = {
+            'id': self.dca.id,
+            'sn': self.dca.sn,
+            'barcode': self.dca.barcode,
+            'hostname': self.dca.hostname,
+            'model': self.dca.model_id,
+            'orientation': self.dca.orientation,
+            'service_env': self.dca.service_env_id,
+            'status': self.dca.status,
+            'depreciation_rate': self.dca.depreciation_rate,
+        }
+        data.update(dca_data or {})
+        if inline_data:
+            data.update(self._prepare_inline_data(inline_data))
+        response = self.client.post(self.dca.get_absolute_url(), data)
+        self.assertEqual(
+            response.status_code,
+            302,
+            (
+                repr(response.context['form'].errors)
+                if response.context and 'form' in response.context else ''
+            )
+        )
+
+    def _prepare_inline_data(self, d):
+        return {
+            '{}{}'.format(self.custom_fields_inline_prefix, k): v
+            for (k, v) in d.items()
+        }
+
+    def test_update_data_center_asset_check_mail_notifications(self):
```

Comment:
What about some more descriptive names for tests in this module..? E.g. "test for ... when ...", "test if ... given ..." etc.